### ISSUE TITLE
Plotting model with second spectrum then removing it does not work

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#3345]
Something goes wrong with the navigator when adding and then removing a second navigator.

Example:

```python
import numpy as np
import hyperspy.api as hs

data = np.random.random((10, 20, 500))
s = hs.signals.Signal1D(data)
m = s.create_model()
m.plot()
```

- Move the red navigator
- Press the `E` key on your keyboard
- Move the new, blue, navigator
- Press the `E` key to remove the second navigator

The red navigator can no longer be moved, and an error is raised: 

```python
matplotlib/axes/_base.py", line 3570, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```

Movie example of this:

[model_plot_bug_second_navigator.webm](https://github.com/hyperspy/hyperspy/assets/1690979/649d311f-870b-48bc-aea0-348a4403b082)

## Python environment:
 - HyperSpy version: current `RELEASE_next_patch`
 - Python version: 3.11

**Comment by ericpre** [comment]
I don't think that this is a supported feature and to avoid the issue, adding another cursor should be disabled when plotting model.
Can you provide the full traceback to figure out where it comes exactly for future reference?

To add such a feature to plot two model in the same figure, it would be necessary to figure out what should the formatting of the lines (signal and model for the different cursor). An simpler alternative would be to display in a separate figure.

**Comment by ericpre** [comment]
Labelling as first good PR, because a simple fix consist in disabling the event for model plotting.

Adding the new feature to plot several model can be done separately.
